### PR TITLE
Update recipe to bump version to 0.9.7

### PR DESCRIPTION
**Pull request by nerdai**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by nerdai** [comment]
<code>@<space/>conda-forge-admin, please rerender</code>

File: recipe/meta.yaml [hunk 1]
```diff
@@ -16,26 +16,41 @@ build:
...
-    - urllib3 <2
-    - fsspec >=2023.5.0
-    - typing_inspect >=0.8.0
-    - typing-extensions >=4.5.0
-    - beautifulsoup4
-    - nest-asyncio
-    - nltk
-    - tree_sitter_languages
+    - python >=3.8<=3.12
+    - SQLAlchemy ={extras = ["asyncio"], version = ">=1.4.49"}
```
Comment (pavelzw) [inline recipe/meta.yaml:23]: this syntax is not supported in conda recipes

try something like

```
    - sqlalchemy >=1.4.49
    - greenlet !=0.4.17
```

see https://github.com/sqlalchemy/sqlalchemy/blob/44771ab0c32730d1ecefac21af13f113b1b19726/setup.cfg#L43

File: recipe/meta.yaml [hunk 2]
```diff
@@ -16,26 +16,41 @@ build:
...
+    - beautifulsoup4 ="^4.12.2"
+    - dataclasses-json ="*"
+    - deprecated =">=1.2.9.3"
+    - fsspec =">=2023.5.0"
+    - httpx ="*"
+    - langchain ={optional = true, version = ">=0.0.303"}
```
Comment (pavelzw) [inline recipe/meta.yaml:29]: this syntax is not supported in conda recipes
you could add it under `run_constrained` or create a separate output similar to https://github.com/conda-forge/matplotlib-feedstock/blob/1fd5929d7f5e3b508cb58ba4e1e9b6cbf5ac3c0c/recipe/meta.yaml#L86-L90

File: recipe/meta.yaml [hunk 3]
```diff
@@ -16,26 +16,41 @@ build:
...
+    - optimum ={extras = ["onnxruntime"], optional = true, version = "^1.13.2"}
+    - sentencepiece ={optional = true, version = "^0.1.99"}
+    - transformers ={extras = ["torch"], optional = true, version = "^4.34.0"}
+    - guidance ={optional = true, version = "^0.0.64"}
+    - lm-format-enforcer ={optional = true, version = "^0.4.3"}
+    - jsonpath-ng ={optional = true, version = "^1.6.0"}
+    - rank-bm25 ={optional = true, version = "^0.2.2"}
+    - scikit-learn ={optional = true, version = "<1.3.0"}
+    - spacy ={optional = true, version = "^3.7.1"}
+    - aiostream ="^0.5.2"
```
Comment (pavelzw) [inline recipe/meta.yaml:52]:
```suggestion
    - aiostream = ">=0.5.2,<0.6.0"
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -16,26 +16,41 @@ build:
...
+    - aiohttp ="^3.8.6"
```
Comment (pavelzw) [inline recipe/meta.yaml:53]:
```suggestion
    - aiohttp = ">=3.8.6,<4.0.0"
```

**Comment by nerdai** [comment]
thanks for the assistance @pavelzw ! I hope I got everything right now.

File: recipe/meta.yaml (hunk 3 above)
Comment (pavelzw) [reply recipe/meta.yaml]: sorry, I meant 

```
    - aiostream >=0.5.2,<0.6.0
```

I fixed it in https://github.com/conda-forge/llama-index-feedstock/pull/20/commits/2ded4da8d4dba0eeb8851813330e5dbf4bb7a654

**Comment by pavelzw** [comment]
@conda-forge-admin, please rerender

**Comment by nerdai** [comment]
thanks @pavelzw !

**Pull request merged**